src/test/java/fi/aalto/cs/apluscourses/utils/CourseFileManagerTest.java

```diff
@@ -0,0 +1,135 @@
+package fi.aalto.cs.apluscourses.intellij.utils;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.io.FileUtilRt;
+import fi.aalto.cs.apluscourses.intellij.model.IntelliJModuleMetadata;
+import fi.aalto.cs.apluscourses.model.ModelExtensions;
+import fi.aalto.cs.apluscourses.model.ModelFactory;
+import fi.aalto.cs.apluscourses.model.Module;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.commons.io.FileUtils;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class CourseFileManagerTest {
+
+  @Test
+  public void testAddEntryForModule() throws IOException, InterruptedException {
```

Comment:
nit: could the internals be grouped and possibly separated into smaller helper methods for better readability?